File: examples/governance/compliance_domain/lib/compliance_domain/training_record/events/assigned_training.rb
module ComplianceDomain
  class TrainingRecord
    module Events
      class AssignedTraining
        attr_reader :aggregate_id, :stakeholder_id, :policy_id, :completed_at, :expires_at, :certification_id, :status, :occurred_at

        def initialize(aggregate_id: nil, stakeholder_id: nil, policy_id: nil, completed_at: nil, expires_at: nil, certification_id: nil, status: nil)
          @aggregate_id = aggregate_id
          @stakeholder_id = stakeholder_id
          @policy_id = policy_id
          @completed_at = completed_at
          @expires_at = expires_at
          @certification_id = certification_id
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/assigned_training.go
package domain

import "time"

type AssignedTraining struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	StakeholderId string `json:"stakeholder_id"`
	PolicyId string `json:"policy_id"`
	CompletedAt time.Time `json:"completed_at"`
	ExpiresAt time.Time `json:"expires_at"`
	CertificationId string `json:"certification_id"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e AssignedTraining) EventName() string { return "AssignedTraining" }

func (e AssignedTraining) GetOccurredAt() time.Time { return e.OccurredAt }
